Disable pytest-services plugin in test to avoid worker_id conflict

mgorny opened the pull request:
Pass "-p no:pytest-services" in test_worker_id_fixture() to ensure
that the presence (and implicit loading) of pytest-services will not
cause the test to fail via overriding worker name.

Fixes #611

Here's a quick checklist that should be present in PRs:

- [ ] Make sure to include reasonable tests for your change if necessary

- [x] We use [towncrier](https://pypi.python.org/pypi/towncrier) for changelog management, so please add a *news* file into the `changelog` folder following these guidelines:
  * Name it `$issue_id.$type` for example `588.bugfix`;
  * If you don't have an issue_id change it to the PR id after creating it
  * Ensure type is one of `removal`, `feature`, `bugfix`, `vendor`, `doc` or `trivial`
  * Make sure to use full sentences with correct case and punctuation, for example:

    ```
    Fix issue with non-ascii contents in doctest text files.
    ```

nicoddemus commented:
Thanks @mgorny!

I changed your PR a bit: I added the `no:pytest-services` flag to `tox.ini` so we don't add a new test in the future reintroducing the problem, and I don't think a changelog is needed because this is a change in our internal testing so users probably are not interested in that. 👍

mgorny commented:
> Thanks @mgorny!
> 
> I changed your PR a bit: I added the `no:pytest-services` flag to `tox.ini` so we don't add a new test in the future reintroducing the problem, and I don't think a changelog is needed because this is a change in our internal testing so users probably are not interested in that.

This doesn't seem to work for me. I suspect that `tox.ini` isn't used by `testdir.runpytest()`.

nicoddemus commented:
Ahhh 🤦 

Sorry, you are right. I will open a new PR later addressing this then, sorry about that!

nicoddemus commented:
Hmm strange, I cannot reproduce the issue with or without the patch: I've tried installing `pytest-services` manually and running pytest, and also added `pytest-services` to `tox.ini` as you suggest and running `tox -e py37-pytestrelease`.

Here's an example:

```
λ pytest testing\acceptance_test.py -k test_worker_id -v
======================================= test session starts =======================================
platform win32 -- Python 3.6.6, pytest-6.2.0.dev265+g902739cfc.d20201212, py-1.9.0, pluggy-0.13.1 -- d:\projects\pytest-xdist\.env36\scripts\python.exe
cachedir: .pytest_cache
rootdir: d:\projects\pytest-xdist, configfile: tox.ini
plugins: forked-1.1.3, services-2.2.1, xdist-2.2.1.dev2+gad99d94.d20201215
collected 90 items / 88 deselected / 2 selected

testing/acceptance_test.py::test_worker_id_fixture[0] PASSED                                 [ 50%]
testing/acceptance_test.py::test_worker_id_fixture[2] PASSED                                 [100%]

================================ 2 passed, 88 deselected in 1.31s =================================
```

(We can see `services-2.2.1` in the header meaning that the plugin is active)

Can you double check your reproduction example, and share any details that might be missing? Thanks!

mgorny commented:
I've actually tested on py36 (tests are run from oldest to newest python). I'll retry with py37 in an hour or so.

mgorny commented:
py37 fails too, although I'm running `py37-pytestlatest` (tox reports: `ERROR: unknown environment 'py37-pytestrelease'`, so I presume that was a typo).

nicoddemus commented:
> presume that was a typo

oops yes. 😁 

Err weird I still cannot reproduce it with `tox -re py37-pytestlatest`.

I see they do declare `worker_id` [here](https://github.com/pytest-dev/pytest-services/blob/8b55085683a65a7f22f9f33419c42d5d798de8b9/pytest_services/service.py#L23-L34), which explains the failure you are seeing, however I don't understand why I can't reproduce it... perhaps because I'm on Windows?

Perhaps we should open an issue on pytest-services to discuss the problem?
